Source/events.handling/EventHandlers.ts

```diff
@@ -44,12 +47,25 @@ export class EventHandlers implements IEventHandlers {
                 }
             }),
             map((value: EventHandlerDecoratedType) => {
-                _logger.debug(`Register EventHandler '${value.eventHandlerId}'`);
-
                 const methodsByArtifact = new ArtifactMap<EventHandlerSignature<any>>();
                 for (const method of HandlesDecoratedMethods.methodsPerEventHandler.get(value.type)!) {
                     const artifact = _artifacts.getFor(method.eventType);
-                    methodsByArtifact.set(artifact, method.method);
+                    methodsByArtifact.set(artifact, (event, eventContext) => {
+
+                        if (method.owner) {
+                            let instance: any;
+                            try {
+                                instance = this._container.get(method.owner);
+                            } catch (ex) {
+                                this._logger.error('Unable to create instance of event handler.', ex);
+                                return new Promise<void>(() => { });
```

Comment:
We need to re-throw the exception so that we retry the event handling later - by swallowing it the event is simply ignored. Also, the logging should probably happen outside this method (if we don't already do that).